### PR TITLE
CODENVY-27; allow to open workspaces by using /ide/namespace/ws_name schema

### PR DESCRIPTION
**Pull request by mshaposhnik**

### CODE REVIEW

File: assembly/assembly-ide-war/src/main/java/org/eclipse/che/DashboardRedirectionFilter.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che;
+
+
+import javax.servlet.*;
```
Comment (sleshchenko) [inline assembly/assembly-ide-war/src/main/java/org/eclipse/che/DashboardRedirectionFilter.java]: We should not use \* in imports

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java [hunk 2]
```diff
@@ -752,6 +750,10 @@ private static String getCurrentUserId() {
         return EnvironmentContext.getCurrent().getSubject().getUserId();
     }
 
+    private static String getCurrentNamespace() {
```
Comment (skabashnyuk) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java]: I would say this method should be inline

File: wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/WorkspaceManagerTest.java [hunk 3]
```diff
@@ -120,7 +121,7 @@ public void setUp() throws Exception {
         EnvironmentContext.setCurrent(new EnvironmentContext() {
             @Override
             public Subject getSubject() {
-                return new SubjectImpl("Test User", USER_ID, "token", new ArrayList<>(), false);
+                return new SubjectImpl(NAMESPACE, USER_ID, "token", new ArrayList<>(), false);
```
Comment (benoitf) [inline wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/WorkspaceManagerTest.java]: do we need arraylist or an empty list ?

Comment (mshaposhnik) [inline wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/WorkspaceManagerTest.java]: Do you imply Collections.emptyList() here ?

Comment (mshaposhnik) [inline wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/WorkspaceManagerTest.java]: We can simply pass null and new emptySet ill be created

**Comment by skabashnyuk** [comment]
Other ok

File: wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/WorkspaceManagerTest.java (hunk 3 above)
Comment (benoitf) [inline wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/WorkspaceManagerTest.java]: Ok

**Comment by vparfonov** [comment]
ok

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java [hunk 4]
```diff
@@ -247,7 +245,7 @@ public void delete(@ApiParam("The workspace id") @PathParam("id") String id) thr
                                                                                         ConflictException,
                                                                                         ForbiddenException {
         if (!workspaceManager.getSnapshot(id).isEmpty()) {
-            machineManager.removeSnapshots(getCurrentUserId(), id);
+            machineManager.removeSnapshots(EnvironmentContext.getCurrent().getSubject().getUserId(), id);
```
Comment (sleshchenko) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java:248]: https://github.com/eclipse/che/blob/master/wsmaster/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/dao/SnapshotDao.java#L78-78
namespace should be equal to user name, no?

**Pull request merged**